agent/consul/state/acl.go

```diff
@@ -2107,48 +1823,50 @@ func (s *Store) ACLAuthMethodList(ws memdb.WatchSet) (uint64, structs.ACLAuthMet
 	}
 
 	// Get the table index.
-	idx := maxIndexTxn(tx, "acl-auth-methods")
+	idx := s.aclAuthMethodMaxIndex(tx, nil, entMeta)
 
 	return idx, result, nil
 }
 
-func (s *Store) ACLAuthMethodDeleteByName(idx uint64, name string) error {
-	return s.aclAuthMethodDelete(idx, name, "id")
+func (s *Store) ACLAuthMethodDeleteByName(idx uint64, name string, entMeta *structs.EnterpriseMeta) error {
+	return s.aclAuthMethodDelete(idx, name, entMeta)
 }
 
-func (s *Store) ACLAuthMethodBatchDelete(idx uint64, names []string) error {
+func (s *Store) ACLAuthMethodBatchDelete(idx uint64, names []string, entMeta *structs.EnterpriseMeta) error {
 	tx := s.db.Txn(true)
 	defer tx.Abort()
 
 	for _, name := range names {
-		s.aclAuthMethodDeleteTxn(tx, idx, name, "id")
+		// NOTE: it may seem odd that one EnterpriseMeta is being used for all of the auth methods being
+		// deleted. However we never actually batch these deletions as auth methods are not replicated
+		// Therefore this is fine but if we ever change that precondition then this will be wrong (unless
+		// we ensure all deletions in a batch should have the same enterprise meta)
+		s.aclAuthMethodDeleteTxn(tx, idx, name, entMeta)
 	}
 
-	if err := indexUpdateMaxTxn(tx, idx, "acl-auth-methods"); err != nil {
-		return fmt.Errorf("failed updating index: %v", err)
-	}
+	//
+	// TODO figure out how to modify the batch deletion struct to blow away the auth methods.
```

Comment:
Did you want to use an annotated TODO here?

Reply:
TODO is outdated here.